src/users/users.service.ts

```diff
@@ -21,7 +24,62 @@ export class UsersService {
       password: hashedPassword,
     });
 
-    return this.users.save(user);
+    const confirmationToken = await this.generateUniqueToken();
+
+    const confirmationLink = `http://localhost:3000/confirmation?token=${confirmationToken}`;
```

Comment:
Why localhost? If you run you project on stage, it will be redirect to localhost